lib/ruby-push-notifications/apns/apns_notification.rb

```diff
@@ -67,7 +67,7 @@ def device_token(token)
       # @return [String]. Binary representation of the notification's payload.
       def payload
         @encoded_payload ||= -> {
-            json = JSON.dump(@data).force_encoding 'ascii-8bit'
+            json = (@data.respond_to?(:to_json) ? @data.to_json : JSON.dump(@data)).force_encoding 'ascii-8bit'
```

Comment:
@waqar-baig. I just want to know the issues you have faced, to be sure we really need this here.

Apart from this, are you working on the documentation?